### PR TITLE
[QA only] Isolate QA features from PR #2022

### PR DESCRIPTION
**Pull request by gandrewstone**
Allow parallel_rpc to run from command line, speed up mempool_push.py
Add useful output on a timeout.
Simultaneously spawn bitcoinds
Fix race condition in tests between bitcoind writing to the log, and the test checking it by looping and waiting if the desired log messages were not found.
Fix logging object in mininode
add core file detection and display
add TRAVIS variable to container so scripts know we are under the travis ci build environment
add core analysis file, clean up printouts

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -33,31 +33,45 @@ env:
...
 matrix:
   include:
-    #bitcoind clang (no depend, only system lib installed via apt)
-    - compiler: clang
+    #x86_64 Linux + deps as via system lib
+    - compiler: gcc
       env:
-        - CXX=clang++-5.0 CC=clang-5.0
-        - CXXFLAGS="-std=c++14"
+        - A="ia64 linux  full tests  system deps, enable debug, debug boost"
```
Comment (sickpig) [inline .travis.yml:39]: we can use https://docs.travis-ci.com/user/build-stages#naming-your-jobs-within-build-stages rather than custom env var to name tests.

File: .travis/script_b.sh [hunk 2]
```diff
@@ -11,18 +11,24 @@ fi
 
 cd "build" || (echo "could not enter distdir build"; exit 1)
```
Comment (sickpig) [inline .travis/script_b.sh:13]: nit: What about adding a comment about what we are doing here, maybe it's self explanatory but you never know.

File: qa/pull-tester/rpc-tests.py [hunk 3]
```diff
@@ -255,7 +261,7 @@ def option_passed(option_without_dashes):
     'rpc_getblockstats',
     WhenElectrumFound('electrum_cashaccount'),
     'minimaldata-activation',
-    'schnorrmultisig-activation',
+    'schnorrmultisig_activation',
```
Comment (sickpig) [inline qa/pull-tester/rpc-tests.py:264]: Ouch seems like we didn't run schnorr musig activation since the file name has been renamed. good catch.

File: qa/rpc-tests/miningtest.py [hunk 4]
```diff
@@ -156,15 +156,16 @@ def run_test(self):
         self.nodes[0].sendtoaddress(self.nodes[1].getnewaddress(), 1)
         self.nodes[0].sendtoaddress(self.nodes[1].getnewaddress(), 1)
         self.sync_all()
+        # print(str(self.nodes[0].getmempoolinfo()))
         assert_equal(str(self.nodes[0].getnetworkinfo()["relayfee"]), "0.01000000")
-        self.nodes[0].generate(1);
+        self.nodes[0].generate(1)
         self.sync_all()
-        print(str(self.nodes[0].getmempoolinfo()["mempoolminfee"]))
+        # print(str(self.nodes[0].getmempoolinfo()))
```
Comment (sickpig) [inline qa/rpc-tests/miningtest.py]: ^^ delete intstead?

File: qa/rpc-tests/miningtest.py [hunk 5]
```diff
@@ -173,7 +174,8 @@ def Test():
     t = MyTest()
```
Comment (sickpig) [inline qa/rpc-tests/miningtest.py]: not: while at it why not using something else rather then `MyTest`? what about `MiningTest`?

File: qa/rpc-tests/schnorrmultisig_activation.py [hunk 6]
```diff
@@ -153,6 +153,7 @@ def check_for_ban_on_rejected_block(self, block, reject_reason=None):
             [block], self.nodes[0], success=False, reject_reason=reject_reason, expect_ban=True)
 
     def run_test(self):
+        print("Initializing test directory "+self.options.tmpdir)
```
Comment (sickpig) [inline qa/rpc-tests/schnorrmultisig_activation.py]: `logger.info` instead of `print`?

File: qa/rpc-tests/wallet_bench.py [hunk 7]
```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python3
```
Comment (sickpig) [inline qa/rpc-tests/wallet_bench.py:1]: this file seems unrelated to this PR

File: qa/pull-tester/rpc-tests.py [hunk 8]
```diff
@@ -531,9 +550,56 @@ def comms(timeout):
...
 
-                if proc.poll() is not None:
+                retval = proc.poll()
+                if not retval is None:
                     if not got_outputs[0]:
-                        comms(50)
+                        try:
+                            comms(0.1)
+                        except subprocess.TimeoutExpired as e:
+                            # print("%s: communicate timed out, but process should have been killed already!" % proc.args[0])
```
Comment (sickpig) [inline qa/pull-tester/rpc-tests.py]: nit: remove all commented `print`s

File: qa/rpc-tests/wallet_bench.py (hunk 7 above)
Comment (gandrewstone) [reply qa/rpc-tests/wallet_bench.py:1]: well, its a new QA "test" that benchmarks the wallet... and this PR is kind of a grab bag of QA improvements.

File: .travis.yml (hunk 1 above)
Comment (gandrewstone) [reply .travis.yml:39]: This doc shows how to name jobs, but we are using some different feature  "matrix".  The linked docs mention matrix but not how to name them.  Do you know how?

File: qa/pull-tester/rpc-tests.py (hunk 8 above)
Comment (gandrewstone) [reply qa/pull-tester/rpc-tests.py]: I did it, but these prints both comment the code and also are useful when something goes wrong.  I guess I'm trying to say that I disagree with the idea that we shouldn't leave useful debug code in comments for use by devs who have to maintain the code later.  

In this particular case, its not documented anywhere that a core dump and only a core dump will cause this exact behavior by the python .communicate() function so I was thinking that this code may need to be re-examined in the future.

File: .travis.yml (hunk 1 above)
Comment (gandrewstone) [reply .travis.yml:39]: one thing I like about putting them as env vars is that they show up in the travis synopsis page in a field that's not useful to us because we have so many environment vars that the field just shows the same thing unless you roll over it.

File: .travis/script_b.sh (hunk 2 above)
Comment (gandrewstone) [reply .travis/script_b.sh:13]: done

File: qa/rpc-tests/miningtest.py (hunk 4 above)
Comment (gandrewstone) [reply qa/rpc-tests/miningtest.py]: done

File: qa/rpc-tests/miningtest.py (hunk 5 above)
Comment (gandrewstone) [reply qa/rpc-tests/miningtest.py]: done

File: qa/rpc-tests/schnorrmultisig_activation.py (hunk 6 above)
Comment (gandrewstone) [reply qa/rpc-tests/schnorrmultisig_activation.py]: done (logging, btw)

File: qa/pull-tester/rpc-tests.py (hunk 8 above)
Comment (sickpig) [reply qa/pull-tester/rpc-tests.py]: fair enough

File: .travis.yml (hunk 1 above)
Comment (sickpig) [reply .travis.yml:39]: > This doc shows how to name jobs, but we are using some different feature "matrix". The linked docs mention matrix but not how to name them. Do you know how?

I think that `jobs` and `matrix` are different declination of the same concept and that `name` works in both of them.

That said we could leave it for another PR.

Comment (gandrewstone) [reply .travis.yml:39]: Let's do that so we can get these fixes deployed to the other PRs so the don't fail spuriously as much.

**Review by sickpig — APPROVED** [review]
